slang.h

```diff
@@ -4134,6 +4134,12 @@ namespace slang
         None, UnsizedArray, StructuredBuffer, ConstantBuffer, ParameterBlock
     };
 
+    // A struct for storing checksums designed to be used with the MD5 hashing implementation
+    struct Checksum
+    {
+        uint32_t checksum[4] = { 0 };
+    };
+
```

Comment:
This was mostly added to encapsulate the MD5 hash representation of a 4 element uint32_t array, otherwise all the hashing code was creating and passing around arrays everywhere. I'm fine with a rename to MD5Digest and changing it in the future if need be.